src/server/pkg/errutil/errutil.go

```diff
@@ -1,9 +1,17 @@
 package errutil
 
 import (
+	"fmt"
 	"strings"
 )
 
+var (
+	// ErrBreak is an error used to break out of call back based iteration,
+	// should be swallowed by iteration functions and treated as successful
+	// iteration.
+	ErrBreak = fmt.Errorf("BREAK")
```

Comment:
This is just a note about changing this idiom to returning a break/done boolean, rather than returning a break within an error.

Reply:
:+1: yeah, ErrBreak is pretty confusing and is likely to encourage buggy implementations of iteration methods.

Reply:
So having thought about this a little more I think I like what we have now better.
Having every return line in an iteration function have to specify that it doesn't want to break by returning `false` leads to a lot of clutter, and it prevents some particularly nice cases where we can pass GRPC `Send` methods directly to iteration functions since they have the right signature.
I also did some more reading of the standard library [function](https://golang.org/pkg/path/filepath/#Walk) this approach is based on, and they do a similar thing where they have a special purpose error that the callback can return, to control the flow of iteration.

Reply:
Okay, I am convinced. Although it is still kind of weird, the benefit of not having to wrap the sends and being more concise is probably worth it.